pkg/kubelet/cm/cpumanager/metrics_test.go

```diff
@@ -0,0 +1,143 @@
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/utils/cpuset"
+)
+
+func TestMetricsAllocate(t *testing.T) {
+	// restore the metric after this UT.
+	bakContainerAlignedComputeResources := metrics.ContainerAlignedComputeResources
+	defer func() {
+		metrics.ContainerAlignedComputeResources = bakContainerAlignedComputeResources
+	}()
+
+	metrics.ContainerAlignedComputeResources = metrics2.NewCounterVec(
+		&metrics2.CounterOpts{
+			Subsystem:      metrics.KubeletSubsystem,
+			Name:           metrics.ContainerAlignedComputeResourcesNameKey,
+			Help:           "Cumulative number of aligned compute resources allocated to containers by alignment type.",
+			StabilityLevel: metrics2.ALPHA,
+		},
+		[]string{metrics.ContainerAlignedComputeResourcesScopeLabelKey, metrics.ContainerAlignedComputeResourcesBoundaryLabelKey},
+	)
+	metrics.ContainerAlignedComputeResources.Create(&semver.SpecVersion)
+
+	testCases := []struct {
+		staticPolicyTest
+		expCounterMetricValue float64
+	}{
+		{
+			staticPolicyTest: staticPolicyTest{
+				description: "SocketNoSMT",
+				topo:        topoUncoreDualSocketNoSMT,
+				pod: makeMultiContainerPod(
+					[]struct{ request, limit string }{
+						{"4000m", "4000m"}},
+					[]struct{ request, limit string }{
+						{"2000m", "2000m"}}),
+				containerName:   "initContainer-0",
+				stAssignments:   state.ContainerCPUAssignments{},
+				stDefaultCPUSet: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7),
+			},
+			expCounterMetricValue: 2,
+		},
+		{
+			staticPolicyTest: staticPolicyTest{
+				description: "SingleSocketHT",
+				topo:        topoSingleSocketHT,
+				pod: makeMultiContainerPod(
+					[]struct{ request, limit string }{
+						{"4000m", "4000m"}},
+					[]struct{ request, limit string }{
+						{"2000m", "2000m"}}),
+				containerName:   "initContainer-0",
+				stAssignments:   state.ContainerCPUAssignments{},
+				stDefaultCPUSet: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7),
+			},
+			expCounterMetricValue: 0,
+		},
+		{
+			staticPolicyTest: staticPolicyTest{
+				description: "SocketNoSMT",
+				topo:        topoUncoreDualSocketNoSMT,
+				pod: makeMultiContainerPod(
+					[]struct{ request, limit string }{},
+					[]struct{ request, limit string }{
+						{"1000m", "1000m"}}),
+				containerName:   "initContainer-0",
+				stAssignments:   state.ContainerCPUAssignments{},
+				stDefaultCPUSet: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7),
+			},
+			expCounterMetricValue: 1,
+		},
+		{
+			staticPolicyTest: staticPolicyTest{
+				description: "SingleSocketHT",
+				topo:        topoSingleSocketHT,
+				pod: makeMultiContainerPod(
+					[]struct{ request, limit string }{},
+					[]struct{ request, limit string }{
+						{"1000m", "1000m"}}),
+				containerName:   "initContainer-0",
+				stAssignments:   state.ContainerCPUAssignments{},
+				stDefaultCPUSet: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7),
+			},
+			expCounterMetricValue: 0,
+		},
+	}
+
+	for _, testCase := range testCases {
```

Comment:
should we do `t.Run` or those must run sequentially?

Reply:
Yes, they need to run sequentially ;)

Reply:
please adda comment in code so it will be clear that this is the case. Otherwise it may be added later by mistake